src/rrd_cgi.c

```diff
@@ -560,13 +559,13 @@ static char *rrdsetenv(
     const char **args)
 {
     if (argc >= 2) {
-        const size_t len = strlen(args[0]) + strlen(args[1]) + 2;
-        char *xyz = (char *) malloc(len);
+        char *xyz = NULL;
 
+        asprintf(&xyz, "%s=%s", args[0], args[1]);
```

Comment:
the reason for the existance of the rrd_*print* functions is that they do NOT support locales ...  
normal *print* functions are all locale aware and thus make it hard to produce consistent output.

I did not want to replace the standard *print* functions since we do need locale support in some places ... 

the whole locale thing in this respect is basically about properly outputting numbers ... so here you could easily switch to rrd_asprintf and be done withit.